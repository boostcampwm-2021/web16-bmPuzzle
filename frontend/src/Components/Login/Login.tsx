import * as React from "react";
import GoogleLogin from "react-google-login";

const Login = (props: any) => {
  const google_id: string = process.env.REACT_APP_CLIENT_ID || "";
  const [userObj, setUserObj] = React.useState({
    email: "",
    name: "",
  });

  const onLoginSuccess = async (res: any) => {
    setUserObj({
      ...userObj,
      email: res.profileObj.email,
      name: res.profileObj.name,
    });

    await fetch("http://localhost:5000/api/login", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        id: res.profileObj.name,
      }),
    })
      .then((response) => response.json())
      .then((response) => {
        if (response.code === 200) {
          window.sessionStorage.setItem("id", res.profileObj.name);
        }
      });
  };

  return (
    <div>
      <GoogleLogin
        clientId={google_id}
        buttonText="Google"
        onSuccess={(result) => onLoginSuccess(result)}
        onFailure={(result) => console.log(result)}
      />
    </div>
  );
};

export default Login;
