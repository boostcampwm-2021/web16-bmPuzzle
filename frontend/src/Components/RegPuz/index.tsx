import React from "react";
import styled from "styled-components";
import LevelComponent from "./Components/LevelComponent";
import GlobalStyles from "../../GlobalStyles";
import Header from "../Header";

const RegPuzDiv = styled.div`
  .rp__title {
    display: flex;
  }
  .rp__levelBox {
    display: flex;
  }
`;

const RegPuz: React.FC = () => {
  return (
    <div id="wrapper">
      <Header></Header>
      <div id="body">
        <RegPuzDiv>
          <GlobalStyles />
          <div className="rp__title">
            <p>Title</p>
            <input type="text" name="title" />
          </div>
          <div className="rp__upload">
            <input type="file" name="img" accept="image/*" />
          </div>
          <div className="rp__level">
            <div>Level</div>
            <div className="rp__levelBox">
              <LevelComponent num={1}></LevelComponent>
              <LevelComponent num={2}></LevelComponent>
              <LevelComponent num={3}></LevelComponent>
            </div>
          </div>
          <div className="rp__submit"></div>
        </RegPuzDiv>
      </div>
    </div>
  );
};

export default RegPuz;
