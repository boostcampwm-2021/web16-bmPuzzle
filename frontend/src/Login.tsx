import * as React from "react";
import GoogleLogin from "react-google-login";
import "dotenv/config";

const Login = (props: any) => {
  const google_id: string = process.env.REACT_APP_CLIENT_ID || "";
  const [userObj, setUserObj] = React.useState({
    email: "",
    name: "",
  });

  const onLoginSuccess = (res: any) => {
    setUserObj({
      ...userObj,
      email: res.profileObj.email,
      name: res.profileObj.name,
    });
    window.sessionStorage.setItem("id", res.profileObj.name);
  };
  return (
    <div>
      <GoogleLogin
        clientId={google_id}
        buttonText="Google"
        onSuccess={(result) => onLoginSuccess(result)}
        onFailure={(result) => console.log(result)}
      />
    </div>
  );
};

export default Login;
