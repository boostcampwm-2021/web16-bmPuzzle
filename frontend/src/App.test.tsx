import React from "react";
import { shallow } from "enzyme";
import Login from "@pages/login";
import RegPuz from "@pages/register-puzzle/index";
import Main from "@pages/main/index";
import Mypage from "@pages/mypage/index";
import Ranking from "@pages/ranking/index";
import Warning from "@pages/warning/index";

jest.mock("paper", () => {
  const mpaper: any = {
    setup: () => {
      return true;
    },
    point: () => {
      return true;
    },
  };
  return jest.fn(() => mpaper);
});
describe("renderLogin", () => {
  let component: any;

  it("renders correctly", () => {
    component = shallow(<Login />);
  });

  it("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
describe("renderMain", () => {
  let component: any;

  it("renders correctly", () => {
    component = shallow(<Main />);
  });

  it("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
describe("renderMypage", () => {
  let component: any;

  it("renders correctly", () => {
    component = shallow(<Mypage />);
  });

  it("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
describe("renderRanking", () => {
  let component: any;

  it("renders correctly", () => {
    component = shallow(<Ranking />);
  });

  it("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
describe("renderWarning", () => {
  let component: any;

  it("renders correctly", () => {
    component = shallow(<Warning warn={"noUser"} />);
  });

  it("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
describe("renderRegPuz", () => {
  let component: any;

  it("renders correctly", () => {
    component = shallow(<RegPuz />);
  });

  it("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
