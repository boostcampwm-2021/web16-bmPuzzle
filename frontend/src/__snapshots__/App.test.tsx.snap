// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderLogin matches snapshot 1`] = `ShallowWrapper {}`;

exports[`renderMain matches snapshot 1`] = `ShallowWrapper {}`;

exports[`renderMypage matches snapshot 1`] = `ShallowWrapper {}`;

exports[`renderRanking matches snapshot 1`] = `ShallowWrapper {}`;

exports[`renderRegPuz matches snapshot 1`] = `ShallowWrapper {}`;

exports[`renderWarning matches snapshot 1`] = `ShallowWrapper {}`;
